packages/atomic/src/components/facets/facet-search/facet-search.tsx

```diff
@@ -0,0 +1,53 @@
+import {Component, Event, EventEmitter, h, Prop} from '@stencil/core';
+import {BaseFacetSearchResult} from '@coveo/headless/dist/api/search/facet-search/base/base-facet-search-response';
+
+@Component({
+  tag: 'facet-search',
+  shadow: false,
+})
+export class FacetSearch {
+  @Event() public resultSelected!: EventEmitter<BaseFacetSearchResult>;
+  @Event() public showMoreResults!: EventEmitter<void>;
+  @Event() public facetSearch!: EventEmitter<string>;
+  @Prop() public facetSearchResults!: BaseFacetSearchResult[];
+  @Prop() public moreValuesAvailable!: boolean;
+
+  private get facetSearchInput() {
+    return (
+      <input
+        onInput={(e) =>
+          this.facetSearch.emit((e.target as HTMLInputElement).value)
```

Comment:
Should we limit how often the event is emitted here? Perhaps it could only be emitted when the user has stopped typing?

After clearing my network logs then typing "hello", this is what I see in the dev tools network tab:
![image](https://user-images.githubusercontent.com/54454747/107375844-2f4bfa80-6ab7-11eb-918a-14ab000458cc.png)

Reply:
I would prefer to avoid debouncing and keeping the search-as-you-type UX for facet search. We could always add something at a later point if it's an issue, but I find picking arbitrary timeouts tricky. Too low and it's not reducing the number of requests, too high and people might think Coveo is slow.

Another idea might be to embed this into headless directly rather than at the UI layer, but I'd prefer to wait,